toke.c

```diff
@@ -821,6 +823,43 @@ S_cr_textfilter(pTHX_ int idx, SV *sv, int maxlen)
 }
 #endif
 
+STATIC void
+S_yyerror_non_ascii_message(pTHX_ const U8 * const s)
+{
+    PERL_ARGS_ASSERT_YYERROR_NON_ASCII_MESSAGE;
+
+    yyerror_pv(Perl_form(aTHX_ "Use of non-ASCII character 0x%02X"
+                               " illegal when 'use source::encoding"
+                               " \"ASCII\"' is in effect", *s), 0);
```

Comment:
This message is inconsistent with the POD of `source::encoding`, which spells the import argument `ascii` (in lowercase). I think we should stick to one spelling throughout.